File: backend/Tasque.Core.Common/DTO/Task/TemplateModels/TaskCustomField.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Tasque.Core.Common.Enums;

namespace Tasque.Core.Common.DTO.Task.TemplateModels
{
    public class TaskCustomField
    {
        public string? Name { get; set; }

        public TaskFieldType Type { get; set; }

        public DropdownField? Dropdown { get; set; }

        public List<LabelField>? Labels { get; set; }
    }
}

File: frontend/src/core/models/task/task-template-models/task-custom-field.ts
import { CheckboxField } from './checkbox-field';
import { DropdownField } from './dropdown-field';
import { TaskFieldType } from '../task-field-types';
import { LabelField } from './label-field';

export interface TaskCustomField {
    name: string,
    type: TaskFieldType,
    fieldId?: string,

    labels?: LabelField[],
    dropdown?: DropdownField,
    checkboxes?: CheckboxField[],
}
